File: spring/src/main/java/com/example/demo/web/OrderController.java
package com.example.demo.web;

import com.example.demo.entity.order.Order;
import com.example.demo.service.OrderService;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/orders")
public class OrderController {
    private final OrderService orderService;

    public OrderController(OrderService orderService) {
        this.orderService = orderService;
    }

    @GetMapping
    public List<Order> findAllOrderByLoginUser() {
        return orderService.findAllOrderByLoginUser();
    }

    @GetMapping("/{id}")
    public Order findUserById(@PathVariable Long id) {
        return orderService.findById(id);
    }

    @PostMapping()
    public Order orderProducts(@RequestBody List<Long> productIds) {
        return orderService.orderProducts(productIds);
    }
}

File: spring/src/main/kotlin/com/example/demo/web/OrderController.kt
package com.example.demo.web

import com.example.demo.entity.order.Order
import com.example.demo.service.OrderService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/orders")
class OrderController(
    private val orderService: OrderService
) {
    @GetMapping
    fun findAllOrderByLoginUser(): List<Order> {
        return orderService.findAllOrderByLoginUser()
    }

    @GetMapping("/{id}")
    fun findUserById(@PathVariable id: Long): Order? {
        return orderService.findById(id)
    }

    @PostMapping
    fun orderProducts(@RequestBody productIds: List<Long>): Order? {
        return orderService.orderProducts(productIds)
    }
}
